backend/src/main/java/com/example/vehiclesservice/service/RentingServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.example.vehiclesservice.service;
+
+import com.example.vehiclesservice.api.Renting;
+import com.example.vehiclesservice.api.Vehicle;
+import com.example.vehiclesservice.exception.RentingNotFoundException;
+import com.example.vehiclesservice.exception.VehicleNotFoundException;
+import com.example.vehiclesservice.model.RentingEntity;
+import com.example.vehiclesservice.repository.RentingRepository;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class RentingServiceImpl implements RentingService{
+
+    private final RentingRepository rentingRepository;
+    private final VehicleService vehicleService;
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @Override
+    public Renting startRenting(Renting renting) throws VehicleNotFoundException {
```

Comment:
Think what will happen if two users want to reserve the same vehicle at the same time